### PR TITLE
#43: Added ActivityDTO

### PR DESCRIPTION
**Pull request by wojtekbu**
* Updated  docs/searching_for_conflicts_architecture.png

### CODE REVIEW

File: src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        this.conflictService = conflictService;
+    }
+
+    @Override
+    public List<ActivityDTO> getAllOrganizationActivities(UUID systemAccountId) {
+        List<ActivityDTO> activities = new ArrayList<>();
+        List<OrganizationDTO> orgs = organizationService.findAllWithOwnerId(systemAccountId);
+
+        // TODO: get organization-entities mapping for every organization
+        // to-remove-start
```
Comment (oskarhinc) [inline src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java]: I think this comment line is not neccessary

File: src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        return new PageImpl<>(list, PageRequest.of(currentPage, pageSize), activities.size());
+    }
+
+    private Optional<ActivityDTO> getEntityActivity(UUID orgId, UUID resourceId) {
+        log.debug("Creating Activity for organization: {} with resourceId: {}", orgId, resourceId);
+        try {
+            Optional<OrganizationDTO> opt = organizationService.findOne(orgId);
+            OrganizationDTO org = opt.orElseThrow(() -> new ActivityCreationException(
+                String.format("There are no conflicts for resourceId: %s", orgId)));
```
Comment (oskarhinc) [inline src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java]: "No organization ..."

File: src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+            List<ConflictDTO> conflictDTOS = conflictService.findAllWithResourceIdAndOwnerId(resourceId, org.getAccountId());
+            if (conflictDTOS != null && conflictDTOS.isEmpty()) {
```
Comment (oskarhinc) [inline src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java]: CollectionUtils.isEmpty

File: src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                throw new ActivityCreationException(String.format("There are no conflicts for resourceId: %s", resourceId));
+            }
+
+            return Optional.of(ActivityDTO.builder()
+                .conflicts(conflictDTOS)
+                .organization(org)
+                .build());
+        } catch (ActivityCreationException ex) {
```
Comment (oskarhinc) [inline src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java]: Please do not catch the exception here, but throw it, or return empty() instead

File: src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java (hunk 3 above)
Comment (wojtekbu) [reply src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java]: ok

File: src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java (hunk 4 above)
Comment (wojtekbu) [reply src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java]: ok

File: src/test/java/org/benetech/servicenet/web/rest/ActivityResourceIntTest.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+import static org.benetech.servicenet.web.rest.TestUtil.createFormattingConversionService;
+import static org.benetech.servicenet.web.rest.TestUtil.sameInstant;
+import static org.hamcrest.Matchers.hasItem;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
```
Comment (oskarhinc) [inline src/test/java/org/benetech/servicenet/web/rest/ActivityResourceIntTest.java]: multiple empty line

File: src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        // TODO: get organization-entities mapping for every organization
+        // this mock below will be removed when Organization matching will be available
+        List<UUID> entities = Collections.singletonList(UUID.fromString(EXAMPLE_DATA));
+        UUID orgId = null;
+        if (orgs.get(0) != null) {
+            orgId = orgs.get(0).getId();
+        }
+
+        try {
+            for (UUID entityId : entities) {
```
[the remark this answers is not in the document]
Comment (wojtekbu) [reply src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java]: the exception is thrown when there is no organization identifier provided. All entities are related to a certain organization. try-catch will be surrounded by for loop of the organization later on when we add organizations' matching for entities.

File: src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    private Optional<ActivityDTO> getEntityActivity(UUID orgId, UUID resourceId) throws ActivityCreationException {
+        log.debug("Creating Activity for organization: {} with resourceId: {}", orgId, resourceId);
+        Optional<OrganizationDTO> opt = organizationService.findOne(orgId);
+        OrganizationDTO org = opt.orElseThrow(() -> new ActivityCreationException(
+            String.format("There is no organization for orgId: %s", orgId)));
+
+        List<ConflictDTO> conflictDTOS = conflictService.findAllWithResourceIdAndOwnerId(resourceId, org.getAccountId());
+        if (conflictDTOS == null || conflictDTOS.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (wojtekbu) [reply src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java]: ok

File: src/test/java/org/benetech/servicenet/web/rest/ActivityResourceIntTest.java (hunk 5 above)
Comment (wojtekbu) [reply src/test/java/org/benetech/servicenet/web/rest/ActivityResourceIntTest.java]: ok

File: src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (oskarhinc) [reply src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java]: ok

**Pull request merged**